pkg/sql/opt/norm/rules/select.opt

```diff
@@ -448,3 +448,22 @@ $input
     )
     (RemoveFiltersItem $filter $item)
 )
+
+# NormalizeIsToEq replaces x IS NOT DISTINCT FROM y with x = y 
+# when it can be proved that neither x nor y is NULL and x and y have identical types.
```

Comment:
This should mention that the transformation is not generally valid, it is only valid in the context of a filter because `x = y` may result in `NULL` when `x IS DISTINCT FROM y` does not and `NULL` is falsy in a filter.

Reply:
Also explain that the rule doesn't apply to tuples, which can have surprising behavior with NULL, though it may be possible to include tuples in the future.